src/span.js

```diff
@@ -249,35 +260,97 @@ export default class Span {
-        let tags = [];
-        for (let i = 0; i < this._tags.length; i++) {
-            let tag = this._tags[i];
-            let emptyBuffer = new Buffer(1);
-            emptyBuffer.fill(0);
-            tags.push({
-                key: tag.key,
-                tagType: 'STRING', // TODO(oibe) support multiple tag types
-                vStr: tag.value.toString(),
-                vDouble: 0,
-                vBool: false,
-                vInt16: 0,
-                vInt32: 0,
-                vInt64: 0,
-                vBinary: emptyBuffer
+    static _getTagType(tag: Tag): string {
+        if (tag.value instanceof Number) {
+            return Span._thrift.TagType.DOUBLE;
+        } else if (tag.value instanceof Boolean) {
+            return Span._thrift.TagType.BOOL;
+        } else if (tag.value instanceof Long) { //TODO(oibe) how else to recognize a long?
+            return Span._thrift.TagType.LONG;
+        } else if (tag.value instanceof Buffer) {
+            return Span._thrift.TagType.BUFFER;
+        } else {
+            return Span._thrift.TagType.STRING;
+        }
+    }
+
+    static _getThriftTags(initialTags: Array<Tag>): Array<any> {
+        let thriftTags = [];
+        for (let i = 0; i < initialTags.length; i++) {
+            let tag = initialTags[i];
+
+            let key: string = tag.key;
+            let vType: string = Span._getTagType(tag);
+
+            let vLong: Buffer = Span.emptyBuffer;
+            if (vType === Span._thrift.TagType.LONG) {
+                vLong = tag.value;
+            }
+
+            let vBinary: Buffer = Span.emptyBuffer;
+            if (vType === Span._thrift.TagType.BUFFER) {
+                vBinary = tag.value;
+            }
+
+            let vBool: boolean = false;
+            if (vType === Span._thrift.TagType.BOOL) {
+                vBool = tag.value;
+            }
+
+            let vDouble: number = 0;
+            if (vType === Span._thrift.TagType.DOUBLE) {
+                vDouble = tag.value;
+            }
+
+            let vStr: string = '';
+            if (vType === Span._thrift.TagType.STRING) {
+                vStr = tag.value.toString();
+            }
+
+            thriftTags.push({
+                key: key,
+                vType: vType,
+                vStr: vStr,
+                vDouble: vDouble,
+                vBool: vBool,
+                vLong: vLong,
+                vBinary: vBinary
+            });
+        }
+
+        return thriftTags;
+    }
+
+    _getThriftLogs(): Array<any> {
+        let thriftLogs = [];
+        for (let i = 0; i < this._logs.length; i++) {
+            let log = this._logs[i];
+            thriftLogs.push({
+                'timestamp': log.timestamp,
+                'fields': Span._getThriftTags(log.fields)
             });
         }
 
+        return thriftLogs;
+    }
+
+    _toThrift() {
+        let tags = Span._getThriftTags(this._tags);
+        let logs = this._getThriftLogs();
```

Comment:
can we be consistent and have all _thrift functions static? Also, I would move them to a separate `thrift.js` file